src/common/commonutils/UrlUtils.c

```diff
@@ -71,7 +68,7 @@ char* UrlDecode(char* target)
             }
             else if ('%' == target[j])
             {
-                if (((j + 2) < targetLength) && ('0' == target[j + 1]) && ('A' == toupper(target[j + 2])))
+                if (((j + 2) < targetSize) && ('0' == target[j + 1]) && ('A' == toupper(target[j + 2])))

```

Comment:
Do we need special handling for the newline character?

Reply:
See line 73

Reply:
I mean, the latter else statement decodes hex character in lines 77-81. Isn't it yielding the same result?

Reply:
Please see new version in last iteration.